tests/Model/DataType/LocalizedFieldTest.php

```diff
@@ -92,4 +93,48 @@ public function testLocalizedFieldInsideFieldCollection()
         //new value (de): index 0
         $this->assertEquals('textDE', $loadedItem->getLinput('de'), 'New localized value inside fieldcollection not saved or loaded properly');
     }
+
+    public function testLocalizedFieldFallback()
+    {
+        $configuration = Config::getSystemConfiguration();
+        $configuration['general']['fallback_languages']['de'] = 'en';
+        Config::setSystemConfiguration($configuration);
+
+        $object = TestHelper::createEmptyObject();
+
+        //en values
+        $object->setLinput('TestEN', 'en');
+        $object->setLcheckbox(true, 'en');
+        $object->setLnumber(123, 'en');
+
+        //de values
+        $object->setLinput('TestDE', 'de');
+        $object->setLcheckbox(true, 'de');
+        $object->setLnumber(0, 'de');
+
+        $object->save();
+
+        //empty de values check fallback
+        $object->setLinput('', 'de');
+        $object->setLcheckbox(false, 'de');
+        $object->save();
+
+        $object = DataObject\Unittest::getById($object->getId(), true);
+
+        $this->assertEquals('TestEN', $object->getLinput('de'));
+        $this->assertEquals(0, $object->getLnumber('de')); //fallback should not return, num 0 is set
+
+        //fallback for checkbox is not supported anymore
+        //$this->assertEquals(true, $object->getLcheckbox('de'));
```

Comment:
when value is actively set to `false`, I would not expect that fallback value is taken ... only when value is set to `null`.